src/scheduler/check.c

```diff
@@ -1607,22 +1611,39 @@ check_nodes(status *policy, server_info *sinfo, queue_info *qinfo, resource_resv
 	if (ninfo_arr == NULL || error)
 		return NULL;
 
+	num_nodes = count_array((void **) ninfo_arr);
+	if ((ninfo_arr_filtrd = (node_info **) malloc((num_nodes + 1) * sizeof(node_info *))) == NULL) {
+		log_err(errno, __func__, MEM_ERR_MSG);
+		return NULL;
+	}
+	ninfo_arr_filtrd[0] = NULL;
+
+	for (i = 0, idx = 0; i < num_nodes; i++) {
+		if (!dflt_sched && resresv->is_job && strcmp(ninfo_arr[i]->partition, resresv->job->queue->partition) == 0) {
+			ninfo_arr_filtrd[idx++] = ninfo_arr[i];
+		}
+	}
+	ninfo_arr_filtrd[idx] = NULL;
```

Comment:
Please do this work once and store it on the queue and not once per job.  Doing it once per job will significantly hurt performance.

Reply:
As you suggested now doing the filtering only once at the start of the scheduling cycle.